unlock-app/src/hooks/useProvider.ts

```diff
@@ -167,32 +168,44 @@ export const useProvider = (config: any) => {
     setLoading(false)
   }
 
-  const changeNetwork = async (network: any) => {
+  const changeNetwork = async (networkConf: NetworkConfig | number) => {
+    const networkConfig =
+      typeof networkConf === 'number'
+        ? config.networks[networkConf]
+        : networkConf
+
+    const { id, name } = networkConfig
+
+    // don't change network if not needed
+    if (id === network) {
+      return
+    }
+
     if (provider.isUnlock) {
-      const newProvider = UnlockProvider.reconnect(provider, network)
+      const newProvider = UnlockProvider.reconnect(provider, networkConfig)
       resetProvider(newProvider)
     } else {
       try {
         const changeNetworkRequest = provider.send(
           'wallet_switchEthereumChain',
           [
             {
-              chainId: `0x${network.id.toString(16)}`,
+              chainId: `0x${id.toString(16)}`,
             },
           ],
           account
         )
         await ToastHelper.promise(changeNetworkRequest, {
-          loading: `Changing network to ${network.name}. Please Approve on your wallet.`,
-          error: `Error in changing network to ${network.name}`,
-          success: `Successfully changed network to ${network.name}`,
+          loading: `Changing network to ${name}. Please Approve on your wallet.`,
+          error: `Error in changing network to ${name}`,
```

Comment:
For another PR, I think we should _not_ show this error when the errorCode is `4902` because right now we shouw the error and immediately prompt the user to add the network.